### DISCUSSION TITLE
Wide Screen

### DISCUSSION THREAD

**Discussion by Renhoex** [Techokami/SonicWorldsNext#4] [Q&A]
This was by far the question I got asked about the most during the initial release so I'll just get this out of the way now.

If you want Worlds Next to be in wide screen, go into the menu bar at the top, go to **Project settings**, then look for the **Display** section and go into **Window**, you'll see 2 fields saying "**width**" and "**height**".

Just give width a value that fits the screen resolution you want. Best example is Sonic Mania's 424 x 240

As of writing this you may need to adjust a few elements so that they align with the center of the screen, I'll try to fix these myself over future updates

**Comment by DimensionWarped** [comment]
The Sonic Mania resolution is going to have uneven scaling. You can paper over it with a filter. That said, if the overall level of zoom matching that of the Genesis games matters more to you than perfect visual scaling, it's not a big deal and it's something that becomes less noticeable the higher your monitor's resolution is.

Personally, non-integer scaling drives me batty, so I'd recommend anyone making a release project of a low res godot game to support at least the following two base resolution options --

480x270 - scales to 16:9 1080p/2160p (most HD/4k displays IE televisions) (integer scaling factor 4/8)
512x288 - scales to 16:9 1440p (integer scaling factor 5)

Don't even bother trying to support even scaling on 1366x768... yeah, it's a fairly common resolution for cheap and/or tiny laptops, but it's terrible, nothing scales to it, and anyone using it can put up with a little ugliness. Same goes for other compromise laptop resolutions like 1600x900... requires too much deviation from the norm to get to even scaling.

As for non-standard (IE not 16:9) aspect ratios, some other resolutions you *might* consider supporting... (but probably shouldn't because very few people actually use these aspect ratios and supporting wider aspect ratios *will* have an impact on design of things in a 2D side scroller)

640x270 - scales to WFHD (2560x1080 IE HD ultrawide) (integer scaling factor 4)
688x288 - scales to WQHD (3440x1440 IE QHD ultrawide) (integer scaling factor 5)
1024x288 - scales to DQHD (5120x1440 IE the Samsung Odyssey G9/G9 Neo) (integer scaling 5)

Honestly the only people I see bothering to try to implement any of those are devs who use those specific monitors as their daily driver and are willing to put in extra dev effort to make playing their own games a little more enjoyable.

All of these resolution choices will appear slightly more zoomed out compared to Sonic Mania, but honestly that's a good thing in my book.

**Reply by RileyRosell** [reply]
Personally, I’m fond of 640x360 for “medium resolution” Sprite games. Integer scales perfectly into 720,1080, and 2160. Of course, it doesn’t work as well for adapting pre-existing scales, due to being pretty far away from both 224/240 and 448/480

**Reply by DimensionWarped** [reply]
The only major issue I see with 640x360 is that it's going to look generally zoomed out compared to a normal Sonic game (unless of course you make graphics that fit, but that's a rather substantial undertaking to consider) and that zoom can have a more profound effect on game feel than you might realize.

I kinda like it though and did test it. As is, some camera related stuff needs tweaked to support that resolution, but it's not bad.

EDIT: Oh, and that resolution also has a magical property in that it scales perfectly to 1440p as well (which is becoming the most popular resolution among enthusiasts lately), so consider that a substantial added bonus.

**Reply by RileyRosell** [reply]
Yeah, it's not something that works as well for adapting zooms as much as it does for projects started from the ground up (for example, it also lets you get a similar character height scaling by using a character 60 pixels tall, good for either more subdued character proportions or just for more detail). In hindsight, I suppose that makes it not super relevant to the discussion.
